### PR TITLE
docs: update release guide

### PR DESCRIPTION
**Pull request by gxthrj**
Related to https://github.com/apache/apisix-ingress-controller/issues/146

### CODE REVIEW

File: docs/release-guide.md [hunk 1]
```diff
@@ -174,44 +174,44 @@ $ cd apache-apisix-1.0 && ./utils/check-version.sh 1.0 && cd ..
 $ rm -rf apache-apisix-1.0/.git
 
 # make tar package
-$ tar zcvf apache-apisix-1.0-rc1-src.tar.gz apache-apisix-1.0
+$ tar zcvf apache-apisix-1.0-src.tar.gz apache-apisix-1.0
```
Comment (moonming) [inline docs/release-guide.md]: we already has `makefile` to do this job, don't need this anymore

**Review by moonming — CHANGES_REQUESTED** [review]

File: docs/release-guide.md (hunk 1 above)
Comment (gxthrj) [reply docs/release-guide.md]: Done.

**Review by membphis — APPROVED** [review]

**Comment by juzhiyuan** [comment]
ping @moonming

File: docs/release-guide.md [hunk 2]
```diff
@@ -158,11 +158,11 @@ Committed revision 37434.
...
 ### Make source code package and upload to Apache SVN
 
-Here's an example of preparing a 1.0-rc1 version. Before you make package, make sure you have branch v1.0 ready on github.
+Here's an example of preparing a 1.0 version. Before you make package, make sure you have branch v1.0 ready on github.
 
 ```sh
-# Create a new version number directory and enter, for example: 1.0-rc1
-$ mkdir 1.0-rc1 && cd 1.0-rc1
+# Create a new version number directory and enter, for example: 1.0
+$ mkdir 1.0 && cd 1.0
```
Comment (moonming) [inline docs/release-guide.md]: please add version to ENV, so others can use these cons lines directly

Comment (gxthrj) [reply docs/release-guide.md]: Done

**Review by moonming — APPROVED** [review]

**Pull request merged**